contentcuration/contentcuration/views/nodes.py

```diff
@@ -165,6 +180,153 @@ def get_nodes_by_ids_complete(request, ids):
+    cached_data = cache.get("details_{}".format(node.node_id))
+
+    if cached_data and last_update:
+        last_cache_update = datetime.strptime(json.loads(cached_data)['last_update'], DATE_TIME_FORMAT)
+        if last_update.replace(tzinfo=None) < last_cache_update:
+            return HttpResponse(cached_data)
+
+    # Get resources
+    resources = descendants.exclude(kind=content_kinds.TOPIC)
+
+
+    # Get all copyright holders, authors, aggregators, and providers and split into lists
+    creators = resources.values_list('copyright_holder', 'author', 'aggregator', 'provider')
+    split_lst = zip(*creators)
+    copyright_holders = filter(bool, set(split_lst[0])) if len(split_lst) > 0 else []
+    authors = filter(bool, set(split_lst[1])) if len(split_lst) > 1 else []
+    aggregators = filter(bool, set(split_lst[2])) if len(split_lst) > 2 else []
+    providers = filter(bool, set(split_lst[3])) if len(split_lst) > 3 else []
+
+
+    # Get sample pathway by getting longest path
+    max_level = resources.aggregate(max_level=Max('level'))['max_level']
+    deepest_node = resources.filter(level=max_level).first()
+    pathway = list(deepest_node.get_ancestors()\
+                            .exclude(parent=None)\
+                            .values('title', 'node_id', 'kind_id')
+                ) if deepest_node else []
+    sample_nodes = [
+        {
+            "node_id": n.node_id,
+            "title": n.title,
+            "description": n.description,
+            "thumbnail": get_thumbnail(n),
+        } for n in deepest_node.get_siblings(include_self=True)[0:4]
+    ] if deepest_node else []
+
+    # Get list of channels nodes were originally imported from (omitting the current channel)
+    channel_id = channel and channel.id
+    originals = resources.values("original_channel_id")\
+                        .annotate(count=Count("original_channel_id"))\
+                        .order_by("original_channel_id")
+    originals = {c['original_channel_id']: c['count'] for c in originals}
+    original_channels = Channel.objects.exclude(pk=channel_id)\
+                                    .filter(pk__in=[k for k, v in originals.items()], deleted=False)\
+                                    .values('id', 'name', 'thumbnail', 'thumbnail_encoding')
+    original_channels = [{
+        "id": c["id"],
+        "name": "{}{}".format(c["name"], _(" (Original)") if channel_id == c["id"] else ""),
+        "thumbnail": get_channel_thumbnail(c),
+        "count": originals[c["id"]]
+    } for c in original_channels]
+
+    # Get tags from channel
+    tags = ContentTag.objects.filter(tagged_content__pk__in=descendants.values_list('pk', flat=True))\
+                            .values('tag_name')\
+                            .annotate(count=Count('tag_name'))\
+                            .order_by('tag_name')
+
+    # Get resource variables
+    resource_count = resources.count() or 0
```

Comment:
Wouldn't `resource.count`, unless it's 0, always evaluate to True in a boolean context? And if it's zero, then it's zero anyway?